newIDE/app/src/Utils/GDevelopServices/User.js

```diff
@@ -48,3 +75,68 @@ export const getUserPublicProfile = (
     .get(`${GDevelopUserApi.baseUrl}/user-public-profile/${id}`)
     .then(response => response.data);
 };
+
+const postBadgeIfLocked = (
```

Comment:
I was confused at first about the signification of "locked", can we find something more meaningful?
Should this be `assignBadgeIfNotClaimed` or just `assignBadge`?